library/spdm_responder_lib/libspdm_rsp_algorithms.c

```diff
@@ -383,7 +384,16 @@ libspdm_return_t libspdm_get_response_algorithms(void *context,
                      sizeof(spdm_negotiate_algorithms_request_t) +
                      sizeof(uint32_t) * spdm_request->ext_asym_count +
                      sizeof(uint32_t) * spdm_request->ext_hash_count);
+        alg_type_pre = struct_table->alg_type;
         for (index = 0; index < spdm_request->header.param1; index++) {
+            /* AlgType shall monotonically increase for subsequent entries. */
+            if ((index != 0) && (struct_table->alg_type <= alg_type_pre)) {
```

Comment:
1. `alg_type_pre` needs to be updated inside the for-loop.
2. This code should be moved to the `-=[Validate Response Phase]=-` section at line `287`.

Reply:
1. The update code is already written at line 396.
2. ok